src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1255,25 +1266,36 @@ internal virtual void ApplyMappedFileChanges(SolutionChanges solutionChanges)
 
         private void CheckAllowedSolutionChanges(SolutionChanges solutionChanges)
         {
-            if (solutionChanges.GetRemovedProjects().Any() && !this.CanApplyChange(ApplyChangesKind.RemoveProject))
+            if (!CanApplyChange(ApplyChangesKind.RemoveProject) && solutionChanges.GetRemovedProjects().Any())
```

Comment:
>CanApplyChange [](http://example.com/codeflow?start=17&length=14)

Check if we have capability first as it is faster and non-allocating.

Reply:
Although the changes aren't bad, can we move them into a separate PR?

Reply:
OK.